src/tests/ftest/pool/dynamic_server_pool.py

```diff
@@ -78,42 +66,56 @@ def verify_uuids(self):
         self.expected_uuids.sort()
         self.assertEqual(self.expected_uuids, actual_uuids)
 
+    def check_pool_location(self, hosts, uuid_to_ranks):
+        """Iterate pools and verify that the UUID-named directory exists in
+        each hosts.
+
+        Args:
+            hosts (list of str): Hostnames to search the UUID-named directory.
+            uuid_to_ranks (str to list of int dictionary): UUID to rank list
+                dictionary.
+        """
+        RC_SUCCESS = 0
+        for pool in self.pool:
+            # Note that we don't check mapping between rank and hostname, but it
+            # appears that self.hostlist_servers[0] is always rank0, 1 is rank1,
+            # and the extra server we'll be adding will be rank2.
+            for rank, host in enumerate(hosts):
+                rc = check_for_pool(host, pool.uuid.lower())
+                pool_exists_on_host = rc == RC_SUCCESS
+                # If this rank is in the rank list, there should be the
+                # UUID-named directory; i.e., pool_exist_on_host is True.
+                self.assertEqual(
+                    rank in uuid_to_ranks[pool.uuid.lower()],
+                    pool_exists_on_host)
```

Comment:
check_for_pool returns a return code, so it's better to convert it to True/False. 0 -> True, otherwise -> False.

Reply:
Ah, I see my confusion.  I was thinking of `general_utils.py`'s ['check_pool_files(hosts, uuid)'](https://github.com/daos-stack/daos/blob/master/src/tests/ftest/util/general_utils.py#L430) as opposed to `check_for_pool.py`'s [`check_for_pool(host, uuid)`](https://github.com/daos-stack/daos/blob/master/src/tests/ftest/util/check_for_pool.py#L29).

Reply:
In this case it probably doesn't matter, but if the test ever deals with a larger number of hosts, the `check_pool_files(hosts, uuid)` may be more efficient as it uses clush.  The implementation would have to change slightly to call `check_pool_files()` once for all the hosts where the UUID file is expected to exist and possibly a second time for hosts where the UUID file is expected not to exist.  In this case you may want a new `ServerManager` method where given a list of ranks, it provides a list of hosts.

Also, it may prove beneficial when debugging to always check every host for the UUID file and after the loop fail the test if any of the checks were unsuccessful.

Reply:
I understand your algorithm, but I think the logic becomes simpler if we use check_for_pool here, so I'll keep it as it is unless we want to switch to check_pool_files.
I updated the code to collect the errors first then verify if there was any error at the end.